gtda/homology/tests/test_simplicial.py

```diff
@@ -465,4 +468,101 @@ def test_fp_transform_high_hom_dim(delta):
 def test_fp_fit_transform_plot(X, hom_dims):
     FlagserPersistence(directed=False).fit_transform_plot(
         X_dist, sample=0, homology_dimensions=hom_dims
-        )
+    )
+
+
+X_lsp_cp = coo_matrix((np.array([1, 2, -1, 3, -2, 0.5,
+                                 1, 1, 1, 1, 1, 1]),
+                       (np.array([0, 1, 2, 3, 4, 5,
+                                  0, 1, 2, 3, 4, 0]),
+                        np.array([0, 1, 2, 3, 4, 5,
+                                  1, 2, 3, 4, 5, 5])))
+                      )
+
+diag_lsp_cp = np.array([[-2, 3, 0, -1],
+                        [-1, 2, 0, 1],
+                        [2, -1, 1, 1],
+                        [3, -2, 1, -1]], dtype=float)
+
+
+def test_lsp_fit_transform():
+    lp = LowerStarFlagPersistence(extended=True)
+    result = lp.fit_transform([X_lsp_cp])[0]
+    assert_almost_equal(np.sort(result, axis=0),
+                        np.sort(diag_lsp_cp, axis=0))
+
+
+@composite
+def get_lsp_matrix(draw):
+    """Generate a 1d array of floats, of a given shape. If the shape is not
+    given, generate a shape of at least (4,)."""
+    n_points = draw(integers(3, 10))
+    diag = draw(arrays(dtype=np.float32,
+                       elements=integers(min_value=1, max_value=int(1e2)),
+                       shape=(n_points,), unique=True))
+    n_edges = draw(integers(2, int(n_points*(n_points-1)/2)))
+    list_vertices = lists(integers(min_value=0, max_value=n_points),
+                          min_size=n_edges, max_size=n_edges)
+    edges = draw(lists(list_vertices, min_size=2, max_size=2,
+                       unique_by=tuple(lambda x: x[k] for k in range(n_edges))))
+
+    edges = np.array(edges)
+    X = coo_matrix((np.concatenate([diag, np.ones(n_edges)]),
+                    (np.concatenate([np.arange(n_points), edges[0]]),
+                     np.concatenate([np.arange(n_points), edges[1]]))))
+    return X.toarray()
```

Comment:
This function generates a random matrix, but, currently, it's not used - i don't have ideas for conditions, which would hold for any simplicial complex.

Reply:
We could test equivariance of the diagrams, i.e. that the barcodes are shifted by a constant if all diagonal weights are.

Reply:
Done!